torch/_inductor/decomposition.py

```diff
@@ -684,45 +639,62 @@ def full_like(
         return result.permute(permutation).clone()
 
 
-@register_decomposition(aten.randint_like.default)
-def randint_like(
+def _rand_like(
+    rand_fn: Callable[..., torch.Tensor],
```

Comment:
You can probably use typing_extensions.concatenate + ParamSpec/TypeVarTuple to ensure they support at least the minimum args we need to support for function calls below.